experimental/SetPartitions/src/GenerateCategory.jl

```diff
@@ -0,0 +1,498 @@
+
+        # store pairs as candidates for the tensor products
+        to_tens = Set{Tuple{AbstractPartition, AbstractPartition}}()
+
+        # get all candidates
+        for i in all_partitions
+            # get fitting partitions in advance (improve runtime)
+            all_partitions_temp_tensor = Set()
+            for key in keys(all_partitions_by_size)
+                if size(i) + Int(key) <= max_length
+                    all_partitions_temp_tensor = union(all_partitions_temp_tensor, 
+                        get(all_partitions_by_size, key, -1))
+                end
+            end
+            for ii in all_partitions_temp_tensor
+                if (!((i, ii) in already_t) && (length(upper_points(i)) + 
+                    length(lower_points(i)) + length(upper_points(ii)) + 
+                    length(lower_points(ii)) <= max_length))
+                    push!(to_tens, (i, ii))
+                    push!(already_t, (i, ii))
+                end
+            end
+        end
+        
+        # second phase: all possible tensor products which aren't redundant
+        (all_partitions, already_t, stop_whole, all_partitions_by_size, 
+            all_partitions_by_size_top_bottom, trace) = 
+            do_tensor_products(all_partitions, already_t, to_tens, stop_whole, max_length, 
+            all_partitions_by_size, all_partitions_by_size_top_bottom, trace)
+
+        # add new variations produced by tensor product for composition
+        to_comp = Set{Tuple{AbstractPartition, AbstractPartition}}()
+
+        # get all candidates for
+        for i in all_partitions
+            # get in advance the right second candidate (regarding format)
+            all_partitions_temp_comp = get(all_partitions_by_size_top_bottom[2], 
+                length(upper_points(i)), -1)
+            for ii in all_partitions_temp_comp
+                if !((i, ii) in already_c) && is_composable(i, ii) && 
+                        is_worth_composition(i, ii, max_length)
+                    push!(to_comp, (i, ii))
+                    push!(already_c, (i, ii))
+                end
+            end
+        end
+
+        # third phase: all possible compositions which aren't redundant
+        (all_partitions, already_c, stop_whole, all_partitions_by_size, 
+            all_partitions_by_size_top_bottom, trace) = do_composition(all_partitions, 
+            already_c, stop_whole, max_length, to_comp, all_partitions_by_size, 
+            all_partitions_by_size_top_bottom, trace)
+    end
+
+    # remove all partitions without size n
+    for i in all_partitions
+        if size(i) == n && !(i in all_partitions_of_size_n)
+            push!(all_partitions_of_size_n, i)
+        end
+    end
+
+    if tracing
+        return all_partitions_of_size_n, trace
+    end
+
+    return all_partitions_of_size_n
+end
+
+"""
+    get_trace(trace::Dict, start::Partition)
+
+Print the trace of the partition `start` constructed with `construct_category`.
```

Comment:
trace is any dictionary? Usually I'd expect the function to be called just `trace` as the "get" is automatically implied. But I don't understand the 'Dict'